### PR TITLE
Limit static route result to corresponding vpcId when a gwId is passed

### PR DESCRIPTION
**Pull request by remibergsma**
When a private `gatewayId` is passed, lookup its `VPCid` and return only static routes that belong to this VPC.

### CODE REVIEW

File: cosmic-core/api/src/main/java/org/apache/cloudstack/api/command/user/vpc/ListStaticRoutesCmd.java [hunk 1]
```diff
@@ -90,12 +95,20 @@ public String getCommandName() {
...
             final VpcGateway gateway = _vpcService.getVpcPrivateGateway(gatewayId);
             gatewayCidr.of(NetUtils.ipAndNetMaskToCidr(gateway.getGateway(), gateway.getNetmask()));
         }
         return gatewayCidr;
     }
+
+    private Long retrieveVpcId() {
+        if (gatewayId != null) {
+            final VpcGateway gateway = _vpcService.getVpcPrivateGateway(gatewayId);
+            return gateway.getVpcId();
```
Comment (borisroman) [inline cosmic-core/api/src/main/java/org/apache/cloudstack/api/command/user/vpc/ListStaticRoutesCmd.java:114]: `gateway` can be null here when a non existing private gateway id has been used. Please review @remibergsma

**Review by borisroman — CHANGES_REQUESTED** [review]

File: cosmic-core/api/src/main/java/org/apache/cloudstack/api/command/user/vpc/ListStaticRoutesCmd.java (hunk 1 above)
Comment (remibergsma) [reply cosmic-core/api/src/main/java/org/apache/cloudstack/api/command/user/vpc/ListStaticRoutesCmd.java:114]: Thanks @borisroman will have a look at it

**Comment by remibergsma** [comment]
go build

**Comment by remibergsma** [comment]
go build

**Pull request merged**